lib/dal/src/user.rs

```diff
@@ -137,6 +155,41 @@ impl User {
             .await?;
         Ok(())
     }
+
+    pub async fn invite_to_workspace(
+        ctx: &DalContext,
+        email: &str,
+        workspace_pk: WorkspacePk,
+    ) -> UserResult<()> {
+        if let Some(user) = User::get_by_email_raw(ctx, email).await? {
+            user.associate_workspace(ctx, workspace_pk).await?;
+            return Ok(());
+        }
+
+        // Race condition here, if the user signs-up while we are inviting them the invite will be
+        // lost forever
+        ctx.txns()
+            .await?
+            .pg()
+            .execute(
+                "SELECT user_invite_to_workspace_v1($1, $2)",
+                &[&email, &workspace_pk],
+            )
+            .await?;
```

Comment:
If the DB isn't where the hang occurs, maybe wrap this in a `tokio::time::timeout`?

Reply:
I'm not sure I follow